### PR TITLE
Introduce terraform for deployments to prod cluster with latest app versions

### PR DESCRIPTION
**Pull request by aalsabag**
1. Rode version v0.5.1
2. Rode UI v0.4.0
3. Grafeas Elasticsearch v0.6.2
4. Expose all container versions as top level variables
5. Add ingress class variable to support multiple ingress types
6. Add loadpolicy step to automatically generate a harbor policy
7. Expose namespace variables as top level variables

### CODE REVIEW

File: tf/k8s/variables.tf [hunk 1]
```diff
@@ -21,10 +28,37 @@ variable "update_coredns" {
...
-variable "grafeas_version" {}
+variable "rode_ui_version" {
+  default = ""
+}
+variable "rode_version" {
+  default = ""
+}
+variable "grafeas_version" {
+  default = ""
+}
```
Comment (mrparkers) [inline tf/k8s/variables.tf:46]: I explicitly changed the versions so they weren't optional, which forces you to define them in the root `terragrunt.hcl` file.  I think that's better so it consolidates the list of versions in a single spot.

Comment (mrparkers) [reply tf/k8s/variables.tf:46]: I suppose the `if` statements within the various templates would have to be removed if we do this too.  I still think that's worth it though.

File: .github/workflows/rode-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+      - name: Setup TF
+        uses: hashicorp/setup-terraform@v1
+        with:
+          terraform_version: 0.14.9
+
+      - name: Setup Terragrunt
+        run: |
+          curl -f -Lo terragrunt https://github.com/gruntwork-io/terragrunt/releases/download/v${{env.TERRAGRUNT_VERSION}}/terragrunt_linux_amd64 && \
+          chmod +x terragrunt
```
Comment (mrparkers) [inline .github/workflows/rode-deployment.yaml:38]: I thought that these tools might have already been preinstalled on the github actions runners?  perhaps I'm wrong about this

File: tf/modules/jenkins/agent.tf [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  rule {
+    api_groups = ["policy"]
+    resources = ["poddisruptionbudgets"]
+    verbs = ["*"]
+  }
+  rule {
+    api_groups = ["rbac.authorization.k8s.io"]
+    resources = ["roles", "rolebindings", "clusterroles", "clusterrolebindings"]
+    verbs = ["*"]
+  }
```
Comment (mrparkers) [inline tf/modules/jenkins/agent.tf:66]: this service account probably doesn't need to hand out more rbac, unless there's something I'm missing

File: tf/modules/jenkins/variables.tf [hunk 4]
```diff
@@ -1,3 +1,8 @@
 variable "harbor_namespace" {}
 variable "harbor_host" {}
 variable "jenkins_host" {}
+variable "namespace" {}
+variable "ingress_class" {
+  default = "nginx"
```
Comment (mrparkers) [inline tf/modules/jenkins/variables.tf:6]: I'd set the default to an empty string here.  I don't think ingress classes are required, so an ingress controller with no class wouldn't fulfill this.

File: tf/modules/rode/main.tf [hunk 5]
```diff
@@ -59,6 +59,60 @@ resource "kubernetes_ingress" "rode" {
...
 }
 
+resource "kubernetes_config_map" "policy" {
+  metadata {
+    name = "rode-policy-configmap"
+    namespace  = kubernetes_namespace.rode.metadata[0].name
+  }
+
+  data = {
+//    "loadpolicy.sh" = "${file("${path.module}/loadpolicy.sh")}"
```
Comment (mrparkers) [inline tf/modules/rode/main.tf]: we can probably remove this line

Comment (jknight-liatrio) [reply tf/modules/rode/main.tf]: Good call. Fixed.

File: tf/modules/jenkins/variables.tf (hunk 4 above)
Comment (jknight-liatrio) [reply tf/modules/jenkins/variables.tf:6]: Done.

File: tf/k8s/variables.tf (hunk 1 above)
Comment (jknight-liatrio) [reply tf/k8s/variables.tf:46]: I asked @ChrisSchreiber and his feedback was it should install the default `AppVersion` unless the variables are explicitly set. I believe this is the current behavior and how it's intended to function.

File: .github/workflows/rode-deployment.yaml (hunk 2 above)
Comment (JonRudy) [reply .github/workflows/rode-deployment.yaml:38]: This is something we considered. 
We could create base images with specific tools for the runners then define a huge list in the terraform with specific-labels for each image.
I think this was the quicker solution to have it demo ready that we knew worked.

Comment (ChrisSchreiber) [reply .github/workflows/rode-deployment.yaml:41]: nope not included in the default runner built for the k8s runner controller. we talked about using a custom runner with it pre-installed but there are some other issues to solve first.

File: tf/k8s/variables.tf (hunk 1 above)
Comment (ChrisSchreiber) [reply tf/k8s/variables.tf:46]: if i'm am deploying this to test something I would rather have the versions of things that i'm not testing default to the helm chart version.

**Review by ChrisSchreiber — APPROVED** [review]

**Pull request merged**